Source/NSWindow.m

```diff
@@ -2701,6 +2705,50 @@ - (NSScreen *) screen
   return _screen;
 }
 
+- (void) applicationDidChangeScreenParameters: (NSNotification *)aNotif
+{
+  NSRect oldScreenFrame = [_screen frame];
+  int    screenNumber = [_screen screenNumber];
+  NSRect newScreenFrame;
+  NSRect newFrame;
+
+  // NSLog(@"[NSWindow - %@] Old/New Screen RC: %lu/%lu",
+  //       [self className], [_screen retainCount],
+  //       [[[NSScreen screens] objectAtIndex: 0] retainCount]);
+  
+  // We need to get new screen from renewed screen list because
+  // [NSScreen mainScreen] returns NSScreen object of key window and that object
+  // will never be released.
+  for (NSScreen *scr in [NSScreen screens])
+    {
+      if ([scr screenNumber] == screenNumber)
+        ASSIGN(_screen, scr);
+    }
+  
+  // Do not adjust frame for mini and appicon windows - it's a WM's job. 
+  if ([self isKindOfClass: [NSMiniWindow class]] || self == [NSApp iconWindow])
+    return;
+
+  newScreenFrame = [_screen frame];
+
+  // NSLog(@"[NSWindow - %@] Old/New Screen (%lu): %.0f/%.0f",
+  //       [self className], [_screen retainCount],
+  //       oldScreenFrame.size.width, newScreenFrame.size.width);
+
+  newFrame = _frame;
+  // Screen Y origin change.
+  newFrame.origin.y += newScreenFrame.origin.y - oldScreenFrame.origin.y;
```

Comment:
These calculation seem correct for the case where the origin of a screen changes, what about resolution changes where we should also adjust the size of the window? We have similar code in the restore of a window. Please have a look at that.

Reply:
We don't need to change size of a window if resolution changes. It's not a DPI - it's a pixel dimensions of a screen. So we need to update origin of a window to actual position of a window.